wordpress-mobile/WordPress-Android
# Quick Start: Remove completed list from tasks checklist modal

Parent #16588

This PR makes the following changes to the `Quick Start` tasks checklist modal:

1. Converts `QuickStartFullScreenDialogFragment` to Kotlin and uses `ViewBinding`. [d825ce41cc10879c4e52b2f9aa71864d935f3f82..4d8a21a4381c1b0d9fae524336e6f14d4d79edbd]
2. Removes `Collapsible Completed Tasks Section` from the modal. [cc8bf965b919bcbdf5dc67d5dcf84e4a59fcf025..3d9918a7dde444bb72a15db3dd811bbfcc8b5b52]
3. Removes `Completed View` when all tasks are complete. [28e0426e6a4c4434e4cb647f442cb174171218d0]
4. Simplifies `QuickStartAdapter` and converts it to `ListAdapter`. [557b89d4747ebe9715839368ea04258b5f175ca8]

<img width=600 src="https://user-images.githubusercontent.com/1405144/169815982-06e92639-e601-46a9-a4c2-6755c897fbc0.png"/>

To test:

1. Enable `Quick Start`.
2. Open the `Quick Start` modal.
3. Make sure all tasks are displayed as expected.
4. Complete a task.
6. ✅ Make sure the completed task is greyed out in place.
7. Try skipping a task by long clicking a task.
8. Make sure the task is greyed out.
9. Skip all tasks.
10. ✅ Make sure that all tasks complete screen is no longer displayed.
11. Rotate the device.
12. ✅ Make sure there are no crashes.

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
